src/Stack/Config.hs

```diff
@@ -526,13 +529,20 @@ loadBuildConfig = do
     wanted <- lockCachedWanted stackYamlFP (projectResolver project) $
         fillProjectWanted stackYamlFP config project
 
-    return BuildConfig
-        { bcConfig = config
-        , bcSMWanted = wanted
-        , bcExtraPackageDBs = extraPackageDBs
-        , bcStackYaml = stackYamlFP
-        , bcCurator = projectCurator project
-        }
+    -- Unfortunately redoes getProjectWorkDir, since we don't have a BuildConfig yet
+    workDir <- view workDirL
+    let projectStorageFile = parent stackYamlFP </> workDir </> relFileStorage
+
+    initProjectStorage projectStorageFile $ \projectStorage -> do
```

Comment:
I wonder how much overhead this adds?  `withBuildConfig` appears to be used in a fair number of places (sometimes indirectly), and not all of those actually need the database.  Is it worth having a way to load the build config _without_ making the database available?

Reply:
I'd rather leave that alone for now, and if we see a performance impact in the future we can lazily fill this in.

Reply:
Sounds good.